src/components/swap/GasEstimateBadge.tsx

```diff
@@ -37,15 +36,15 @@ export default function GasEstimateBadge({
   showRoute,
   disableHover,
 }: {
-  trade: InterfaceTrade<Currency, Currency, TradeType> | undefined | null // dollar amount in active chain's stablecoin
+  trade: InterfaceTrade | undefined | null // dollar amount in active chain's stablecoin
   loading: boolean
   showRoute?: boolean // show route instead of gas estimation summary
   disableHover?: boolean
 }) {
   const formattedGasPriceString = trade?.gasUseEstimateUSD
-    ? trade.gasUseEstimateUSD.toFixed(2) === '0.00'
+    ? trade.gasUseEstimateUSD === '0.00'
       ? '<$0.01'
```

Comment:
i would also love to do that, but we do actually parse this value in one place - in `useAutoSlippageTolerance` and it converts it into a CurrencyAmount. so keeping this logic for now

Reply:
rather than include parsing logic here, can this be added as a class method on `ClassicTrade` to keep the logic in one place? eg

```
  gasUseEstimate: CurrencyAmount<X, Y>
  gasUseEstimateDisplay: string
```